### PR TITLE
[com_templates] Rewrote hardcoded SQL to query object

### PR DESCRIPTION
**Pull request by pe7er**
Rewrote 4 SQL queries to query object format.

### CODE REVIEW

File: administrator/components/com_templates/models/style.php [hunk 1]
```diff
@@ -626,20 +626,19 @@ public function setHome($id = 0)
...
 		// Reset the home fields for the client_id.
-		$db->setQuery(
-			'UPDATE #__template_styles' .
-			' SET home = \'0\'' .
-			' WHERE client_id = ' . (int) $style->client_id .
-			' AND home = \'1\''
-		);
+		$query = $db->getQuery(true)
+			->update('#__template_styles')
+			->set('home = 0')
```
Comment (alikon) [inline administrator/components/com_templates/models/style.php]: the `home` field is declared as char/varchar in the `#__template_styles` table so should be quoted as a string

**Review by alikon — CHANGES_REQUESTED** [review]

**Comment by pe7er** [comment]
Thanks your correction @alikon !
I've changed it in the SQL.

**Comment by alikon** [comment]
sorry to be pedantic `$db->quote('0')`   sounds better

**Comment by pe7er** [comment]
@alikon No worries, you're right! 
Using $db->q() is better. I've corrected it.
Thanks!

**Review by alikon — APPROVED** [review]

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 64e9599db1b6c4465523a019a7b9c4ac04d054f1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17921">issues.joomla.org/tracker/joomla-cms/17921</a>.</sub>

File: administrator/components/com_templates/models/style.php [hunk 2]
```diff
@@ -626,20 +626,19 @@ public function setHome($id = 0)
...
-		$db->setQuery(
-			'UPDATE #__template_styles' .
-			' SET home = \'0\'' .
-			' WHERE client_id = ' . (int) $style->client_id .
-			' AND home = \'1\''
-		);
+		$query = $db->getQuery(true)
+			->update('#__template_styles')
+			->set('home = ' .  $db->q('0'))
+			->where('client_id = ' . (int) $style->client_id . ' AND home =  ' . $db->q('1'));
```
Comment (Quy) [inline administrator/components/com_templates/models/style.php]: Move this `'home =  ' . $db->q('1')` into a separate where clause.

**Comment by ghost** [comment]
@alikon can you please inform how you have tested so we can get the needed second Test?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17921">issues.joomla.org/tracker/joomla-cms/17921</a>.</sub>

**Comment by fancyFranci** [comment]
I have tested this item :white_check_mark: successfully on 64e9599db1b6c4465523a019a7b9c4ac04d054f1<br /><br />Switching the default template style is working like before.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17921">issues.joomla.org/tracker/joomla-cms/17921</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Pull request merged**